### PR TITLE
Improve environments pipeline edit

### PR DESCRIPTION
**Pull request by maheshp**
Fixes Issue: #8127

### CODE REVIEW

File: api/api-internal-pipeline-groups-v1/src/main/java/com/thoughtworks/go/apiv1/internalpipelinegroups/representers/InternalPipelineGroupsRepresenter.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                originWriter.add("id", ((RepoConfigOrigin) origin).getConfigRepo().getId());
+            });
+        }
+    }
+
+    public static void toJSON(OutputWriter outputWriter, PipelineGroupsViewModel pipelineGroupsViewModel, Collection<String> users, Collection<String> roles) {
+        toJSON(outputWriter, pipelineGroupsViewModel);
+        outputWriter.addChild("additional_info", (writer) -> writer.addChildList("users", users)
+                .addChildList("roles", roles));
+    }
```
Comment (kritika-singh3) [inline api/api-internal-pipeline-groups-v1/src/main/java/com/thoughtworks/go/apiv1/internalpipelinegroups/representers/InternalPipelineGroupsRepresenter.java:72]: This is not required as the envs page does not need the users and roles info.

Comment (maheshp) [reply api/api-internal-pipeline-groups-v1/src/main/java/com/thoughtworks/go/apiv1/internalpipelinegroups/representers/InternalPipelineGroupsRepresenter.java]: Done

**Comment by kritika-singh3** [comment]
Tested on `GoCD Version: 20.5.0 (11818-b3932994090f442d17d0829804f44554ad04d309)` with 
```
 "Number of pipelines": 4250,
 "Number of environments": 19,
 "Number of agents": 40,
 "Number of unique materials": 3234,
 "Number of schedulable materials": 2734
```

The environments page took around 1-2 sec to render. On clicking the edit pipeline for one of the env, the API call took 1.2 sec and the modal was loaded in total of 2 sec post click.

**Comment by arvindsv** [comment]
@kritika-singh3 Do we have the "before" numbers for this specific instance?

**Comment by kritika-singh3** [comment]
I don't remember the exact duration but it was in the range of 10-20 secs for rendering of the modal for edit pipelines-environment association

**Pull request merged**